hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -383,4 +369,55 @@ private void handlePutOMKeyEvent(String key, OmKeyInfo omKeyInfo,
     }
   }
 
+  /**
+   * Write an OM key to container DB and update containerID -> no. of keys
+   * count.
+   *
+   * @param key key String
+   * @param omKeyInfo omKeyInfo value
+   * @throws IOException if unable to write to recon DB.
+   */
+  private void writeOMKeyToContainerDB(String key, OmKeyInfo omKeyInfo)
+      throws IOException {
+    long containerCountToIncrement = 0;
+    for (OmKeyLocationInfoGroup omKeyLocationInfoGroup : omKeyInfo
+        .getKeyLocationVersions()) {
+      long keyVersion = omKeyLocationInfoGroup.getVersion();
+      for (OmKeyLocationInfo omKeyLocationInfo : omKeyLocationInfoGroup
+          .getLocationList()) {
+        long containerId = omKeyLocationInfo.getContainerID();
+        ContainerKeyPrefix containerKeyPrefix = new ContainerKeyPrefix(
+            containerId, key, keyVersion);
+        if (reconContainerMetadataManager.getCountForContainerKeyPrefix(
+            containerKeyPrefix) == 0) {
+          // Save on writes. No need to save same container-key prefix
+          // mapping again.
+          reconContainerMetadataManager.storeContainerKeyMapping(
+              containerKeyPrefix, 1);
+
+          // check if container already exists and
+          // increment the count of containers if it does not exist
+          if (!reconContainerMetadataManager.doesContainerExists(containerId)) {
+            containerCountToIncrement++;
```

Comment:
I think in earlier case containerKeyMap keep appending and increasing memory during reprocess for each key.
After this change we are querying from rocksdb always, I think better we can avoid that.
May be we can keep map in this function locally to keep containerId so that we can avoid querying rocksdb again for same container. There won't be many containers per key and hence memory will not increase much.

Reply:
Thank you for the review. Since `containerKeyMap` is the main culprit, maybe we can reuse the `containerKeyCountMap` introduced in HDDS-6783, accumulate it during the whole reprocess and persist it at the end as a batch by reusing the `writeToTheDB` function (`containerKeyMap` argument can be left as an empty map).

Reply:
Using `containerKeyCountMap`:
```
if (!containerKeyCountMap.containsKey(containerId)) {
  containerCountToIncrement++;
}
```

Reply:
Yes makes sense.

Reply:
I have tested the new patch, there is no regression and there was significant improvement `ContainerKeyMapperTask#reprocess` processing time.